### PR TITLE
Use new cron plugin for Werft cron jobs

### PR DESCRIPTION
**Pull request by mads-hartmann**
## Description

This moves our existing Werft cron jobs to use the cron plugin which was released in Werft 0.2.0. This greatly simplifies the process for adding and updating Werft cron jobs as you no longer have to modify the Werft config and restart the service when adding or modifying the cron expressions for a cron job.

As Werft only detects jobs in `.werft` I had to move the existing cron jobs to `.werft` - previously this wasn't an issue as the cron definitions for each job in the Werft config pointed to a specific file. But now we're relying on Werft detecting the jobs in order to see the cron expression, so the job specifications need to live directly in `.werft`.

I introduced the same naming convention as we have in gitpod-io/ops, which is `<team name>-<job-name>` and updated CODEOWNERS.

I deleted the `delete-preview-environments` job and instead changed the `delete-preview-environments-cron` job to run every hour. Werft never triggered `delete-preview-environments` when branches were deleted so we have never seen the job work. Cleaning preview environments every hour is good enough for now.

## Related Issue(s)

Fixes https://github.com/gitpod-io/ops/issues/1255

## How to test

Once merged we have to verify that Werft does indeed schedule the cron jobs. As the delete preview environments cron job now runs every hour I will use that one to verify that it gets triggered.

## Release Notes

```release-note
NONE
```

## Documentation

I have to updated our How-to guides on how to add cron jobs.

### CODE REVIEW

**Comment by roboquat** [comment]
@mads-hartmann: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Review by ArthurSens — APPROVED** [review]
I'm adding the hold label just because it's not clear to me if you want this merged straight away or want to prepare something to test if right after merge 🤔

File: .werft/ide/integration-tests-startup.yaml [hunk 1]
```diff
@@ -90,4 +90,7 @@ pod:
           echo "integration test failed" | werft log slice "test"
           exit 1
       fi
-      echo "integration test success" | werft log slice "test"
\ No newline at end of file
+      echo "integration test success" | werft log slice "test"
+plugins:
+  # At 00:30 every day
+  cron: "0 3 * * *"
```
Comment (felladrin) [inline .werft/ide/integration-tests-startup.yaml:96]: Just to confirm: Isn't it "_At 03:00 every day_"? [[1](https://crontab.guru/#0_3_*_*_*)]

**Comment by mads-hartmann** [comment]
@ArthurSens No additional things to prepare, so just have to get it merged so I can see if Werft does indeed respect these new cron expressions ☺️

File: .werft/ide/integration-tests-startup.yaml (hunk 1 above)
Comment (mads-hartmann) [reply .werft/ide/integration-tests-startup.yaml:96]: Haha, you are right! 😝  I will just remove the comment all together and people can paste into https://crontab.guru/

**Review by ArthurSens — APPROVED** [review]

**Review by felladrin — APPROVED** [review]
Thanks for separating it into atomic commits. Made it easy to follow the changes!
Good to go!

**Pull request merged**